oecophylla/cli/launch.py

```diff
@@ -0,0 +1,190 @@
+              default=None,
+              help='Sample sheets used to demultiplex the Illumina run.')
+@click.option('--params', '-p', type=click.Path(exists=True), required=True,
+              help='Specify parameters for the tools in a YAML file.')
+@click.option('--envs', '-e', type=click.Path(exists=True), required=True,
+              help='Specify environments for the tools in a YAML file.')
+@click.option('--cluster-params', type=click.Path(), required=False,
+              help='File with parameters for a cluster job.')
+@click.option('--local-scratch', type=click.Path(), default='/tmp',
+              help='Temporary directory for storing intermediate files.')
+@click.option('--workflow-type',
+              type=click.Choice(['torque', 'slurm', 'local']),
+              default='local',
+              help='Temporary directory for storing intermediate files.')
+@click.option('--log-dir', type=click.Path(), required=False,
+              default='/dev/null',
+              help='Directory containing log files.')
+@click.option('--output-dir', '-o', type=click.Path(),
+              help='Input directory of all of the samples.')
+@click.option('--force', is_flag=True, default=False,
+              help='Restarts the run and overwrites previous input.')
+@click.option('--latency-wait', type=click.INT, default=90,
+              help='Latency wait between cluster jobs.')
+def workflow(input_dir, sample_sheet, params, envs, cluster_params,
+             local_scratch, workflow_type, log_dir, output_dir, force,
+             latency_wait):
+
+    import snakemake
+    from skbio.io.registry import sniff
+
+    # SNAKEMAKE
+    snakefile = "%s/../Snakefile" % os.path.abspath(__file__)
+
+    # INPUT DIR
+    for inp_file in glob.glob('%s/*' % input_dir):
+
+        file_format = sniff(inp_file)[0]
+
+        if (file_format != 'fasta') and (file_format != 'fastq'):
+            raise TypeError('Input files need to be in FASTA or FASTQ format.')
+
+    # OUTPUT
+    # create output directory, if does not exist
+    _create_dir(output_dir)
+
+    # SAMPLE SHEET
+    # If a manifest is not specified, the files containing forward and reverse reads
+    # will be automatically identified using regex.
+    if sample_sheet:
+        _sheet = read_sample_sheet(sample_sheet)
+        sample_dict = extract_samples_from_sample_sheet(_sheet, input_dir)
+    else:
+        sample_dict = extract_sample_paths(input_dir)
+
+    # PARAMS
+    with open(params, 'r') as f:
+        params_dict = yaml.load(f)
+
+    # ENVS
+    with open(envs, 'r') as f:
+        envs_dict = yaml.load(f)
+
+    # CONFIG
+    # merge PARAMS, SAMPLE_DICT, ENVS
+    config_dict = {}
+    config_dict['samples'] = sample_dict
+    config_dict['params'] = params_dict
+    config_dict['envs'] = envs_dict
+
+    config_yaml = yaml.dump(config_dict, default_flow_style=False)
+    config_fp = '%s/%s' % (local_scratch, 'config.yaml')
```

Comment:
would it be possible to add a test near the beginning of the function which verifies that `local_scratch` is in the path, and writable?